cli/pkg/stack_mgr/stack_service.go

```diff
@@ -417,3 +423,87 @@ func (s *StackService) getDistributedLock() (*backend.DistributedLock, error) {
 	lockConfig := backend.DistributedLockConfig{DynamodbTableName: s.backend.Conf().GetDynamoLocktableName()}
 	return backend.NewDistributedLock(&lockConfig, s.backend.GetDynamoDBClient())
 }
+
+func (s *StackService) Generate(ctx context.Context) error {
```

Comment:
This seems quite a bit different than the other functions in `StackService`, is there another place this could live?

Reply:
We have workspaceRepo, orchestrator, stackService.  None of them are a perfect fit. I can create completely new tooling for it...

Reply:
I think something new would be good. As you get into modifying existing HCL I think this will grow and make StackService more multipurpose than it should be